src/variables-store.ts

```diff
@@ -0,0 +1,51 @@
+import { InstanceBase } from '@companion-module/base'
+import { bufferTime, filter, map, mergeMap, Observable, Subject, takeUntil } from 'rxjs'
+
+import { type UiConfig } from './config.js'
+
+export class UiVariablesStore {
+	private variableStreams$ = new Subject<{ variableId: string; stream$: Observable<string | number> }>()
+	private destroy$ = new Subject<void>()
+
+	constructor(private instance: InstanceBase<UiConfig>) {
+		// Subscribe to all variable streams and batch update Companion variables
+		this.variableStreams$
+			.pipe(
+				mergeMap(({ variableId, stream$ }) => {
+					return stream$.pipe(
+						map((value) => ({ variableId, value })),
+						takeUntil(this.destroy$),
+					)
+				}),
+				bufferTime(50), // buffer timeframe for batch updates
```

Comment:
This pipeline merges all variable update streams and buffers them in a 50 ms timeframe to do batch updates.
E.g. if two channels update at the same time, their updates will be batched into one `setVariableValues` call.
`reduce` below overrides values that changed multiple times within the timeframe.

Is it okay to update variables 20 times/second? (It's an audio mixer, so I don't expect too many updates over time)